### DISCUSSION TITLE
Multiphysics simulation to get the acoustic noise

### DISCUSSION THREAD

**Discussion by vinayPrakash31** [Eomys/pyleecan#497] [General]
Hello,

Could you please tell if the multiphysics capability already included in Pyleecan? I haven't been able to find the structural coupling to get the natural frequencies of the stator core and ultimately the radiated acoustic noise. I believe it's not yet implemented and therefore I am trying to do it myself using simple analytical equations (I dont' wanna use another FEA software for the same). Am I missing out on something here?

Thanks for the response.

**Comment by BonneelP** [comment]
Hello,

Thank you for your interest in Pyleecan. 
For now we have Electrical, Magnetics, Forces, Losses models and one Structural model based on Elmer (FEA) but we haven't yet added an Acoustic model. You can add your own structural and acoustic models by adding some new objects that follow the global organisation of the other models (pyleecan was designed to easily add new models and to provide a simple way to compare the results). We can help you to organize your code for that purpure and guide you in the implementation.

For a good start the first questions are:
- Which models do you have in mind ?
- Do you have any publications/reference book that you will use ?
- Do you have some validation data to check the implementation of these models ?
- Which Input / Output are required for your models and how to organized them ?

Best regards,
Pierre

**Reply by vinayPrakash31** [reply]
Hi @BonneelP,
Thanks for your reply. I have already started implementing the structural and the acoustic modules using analytical formulations, however it's only for my PhD research work (Vibroacoustic metamodels) and not strictly implemented as per the coding "convection" of this project. 

Currently, I am working on a simplified typical  IPMSM (customized) and I'm referring a few standard textbooks (for eg., Noise of polyphase electric motors, Gieras et al.) and some other thesis and research articles from J.L.Besnerais, R.Pile and so on. The focus of my research is not to delve deep into the accurate physical modelling aspect of e-motors, and rather to use Pyleecan for e-motors vibroacoustic analysis. Being an open-source project, this aligns well with my research workflow and I would be interested in contributing towards the development with some guidance.

Do you think we can use/include Pyleecan in an industrial toolchain for e-motor noise evaluation? Have there already been  some validation studies performed in the past using Pyleecan?

Thanks again for your reply :)

Best regards,
Vinay

**Comment by BonneelP** [comment]
Hello, 
I would recommend to follow pyleecan coding convention as soon as possible to ease the integration process at the end. Anyway most of the conventions and the tools we use are there to clarify and ease the development of the models.

Thanks to Apache licence there is no restriction to use pyleecan in a commercial way (its something we explained in our [ICEM 2018 publication](https://pyleecan.org/icem.2018.html). In particular, the thesis / research articles you are referring to enables to build [Manatee commercial  software](https://manatee-software.com/) to improve e-NVH performances of electrical machines (J.L.Besnerais, E. Devillers and R.Pile are my colleagues). Sadly, there are very few electromagnetic validation cases on Pyleecan, but I can confirm that it is possible to use it within a validated e-NVH toolchain since this has been done for Manatee ;)

Best Regards,
Pierre
